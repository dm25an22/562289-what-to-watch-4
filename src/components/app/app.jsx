import React, {PureComponent} from "react";
import Main from "../main/main.jsx";
import MovieDetails from "../movie-details/movie-details.jsx";
import PropTypes from "prop-types";
import {Route, Switch, Router} from "react-router-dom";
import {connect} from "react-redux";
import {getFilms, getPromoFilm} from "../../reducer/data/selectors";
import {getAuthStatus, getUserData} from "../../reducer/user/selectors";
import SignIn from "../sign-in/sign-in.jsx";
import {Operation as UserOperation, AuthorizationStatus} from "../../reducer/user/user";
import {Operation as DataOperation} from "../../reducer/data/data";
import {AppRoute} from "../../consts";
import {history} from "../../history";
import withCurrentFilm from "../../hocks/with-current-film/with-current-film.js";
import MyList from "../my-list/my-list.jsx";
import AddReview from '../add-review/add-review.jsx';
import withErrorStyle from "../../hocks/with-error-style/with-error-style";
import VideoFullScreen from "../video-full-screen/video-full-creen.jsx";
import withFullVideoPlayer from "../../hocks/with-full-video-player/with-full-video-player";

const MovieDetailsWrraped = withCurrentFilm(MovieDetails);
const AddReviewWrraped = withCurrentFilm(withErrorStyle(AddReview));
const VideoFullScreenWrraped = withCurrentFilm(withFullVideoPlayer(VideoFullScreen));
class App extends PureComponent {
  render() {
    const {
      films,
      promoFilm,
      authStatus,
      onSubmitAuth,
      loadFavoriteList,
      onSubmitAddReview,
      userData
    } = this.props;

    if (films === null || promoFilm === null || userData === null) {
      return null;
    }

    if (authStatus === AuthorizationStatus.AUTH) {
      loadFavoriteList();
    }

    return (
      <Router history={history}>
        <Switch>
          <Route exact path={AppRoute.ROOT}
            render={() => (
              <Main
                promoFilm={promoFilm}
              />
            )}
          >
          </Route>
          <Route exact path={AppRoute.LOGIN}
            render={() => (
              <SignIn
                onSubmit={onSubmitAuth}
                authStatus={authStatus}
              />
            )}
          >
          </Route>
          <Route exact path={`${AppRoute.FILM}/:id`}
            render={(props) => (
              <MovieDetailsWrraped
                {...props}
                films={films}
                authStatus={authStatus}
              />
            )}
          >
          </Route>
          <Route exact path={AppRoute.MY_LIST}>
            <MyList />
          </Route>
          <Route exact path={`${AppRoute.FILM}/:id${AppRoute.ADD_REVIEW}/:id`}
            render={(props) => (
              <AddReviewWrraped
                {...props}
                films={films}
                onSubmitAddReview={onSubmitAddReview}
              />
            )}
          />
          <Route exact path={`${AppRoute.PLAYER}/:id`}
            render={(props) => (
              <VideoFullScreenWrraped
                {...props}
                films={films}
              />
            )}
          />
        </Switch>
      </Router>
    );
  }

}

App.propTypes = {
  authStatus: PropTypes.string.isRequired,
  films: PropTypes.any,
  promoFilm: PropTypes.any,
  onSubmitAuth: PropTypes.func.isRequired,
  loadFavoriteList: PropTypes.func.isRequired,
  onSubmitAddReview: PropTypes.func.isRequired,
  userData: PropTypes.any
};

const mapStateToProps = (state) => {
  return {
    films: getFilms(state),
    promoFilm: getPromoFilm(state),
    authStatus: getAuthStatus(state),
    userData: getUserData(state)
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onSubmitAuth(authData) {
      dispatch(UserOperation.login(authData));
    },

    loadFavoriteList() {
      dispatch(DataOperation.loadFavorites());
    },

    onSubmitAddReview(filmId, newComment, onSuccess, onError) {
      dispatch(DataOperation.addNewComment(filmId, newComment, onSuccess, onError));
    }
  };
};

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
