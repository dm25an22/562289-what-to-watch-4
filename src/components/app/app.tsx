import * as React from "react";
import Main from "../main/main";
import MovieDetails from "../movie-details/movie-details";
import {Route, Switch, Router} from "react-router-dom";
import {connect} from "react-redux";
import {getFilms, getPromoFilm, getUniqueGeners} from "../../reducer/data/selectors";
import {getAuthStatus, getUserData} from "../../reducer/user/selectors";
import SignIn from "../sign-in/sign-in";
import {Operation as UserOperation} from "../../reducer/user/user";
import {Operation as DataOperation} from "../../reducer/data/data";
import {AppRoute} from "../../enum";
import {history} from "../../history";
import withCurrentFilm from "../../hocks/with-current-film/with-current-film";
import MyList from "../my-list/my-list";
import AddReview from '../add-review/add-review';
import withErrorStyle from "../../hocks/with-error-style/with-error-style";
import VideoFullScreen from "../video-full-screen/video-full-screen";
import withFullVideoPlayer from "../../hocks/with-full-video-player/with-full-video-player";
import withValidateSignIn from '../../hocks/with-validate-sign-in/with-validate-sign-in';
import PrivateRoute from "../private-route/private-route";
import PageWarring from "../page-warring/page-warring";
import NotFoundMessage from "../page-not-found/page-not-found";
import {filmType, userDataType} from "../../types";

const MovieDetailsWrraped = withCurrentFilm(MovieDetails);
const AddReviewWrraped = withCurrentFilm(withErrorStyle(AddReview));
const VideoFullScreenWrraped = withCurrentFilm(withFullVideoPlayer(VideoFullScreen));
const SignInWrraped = withValidateSignIn(SignIn);

interface Props {
  films: filmType[],
  promoFilm: filmType,
  authStatus: string,
  onSubmitAuth: () => void,
  onSubmitAddReview: () => void,
  genres: string[],
  userData: userDataType,
}

const App: React.FC<Props> = (props: Props) => {
  const {
    films,
    promoFilm,
    authStatus,
    onSubmitAuth,
    onSubmitAddReview,
    genres
  } = props;

  return (
    <Router history={history}>
      <Switch>
        <Route exact path={AppRoute.ROOT}
          render={() => (
            <Main
              promoFilm={promoFilm}
              films={films}
              genres={genres}
            />
          )}
        >
        </Route>
        <Route exact path={AppRoute.LOGIN}
          render={() => (
            <SignInWrraped
              authStatus={authStatus}
              onSubmit={onSubmitAuth}
            />
          )}
        >
        </Route>
        <Route exact path={`${AppRoute.FILM}/:id`}
          render={(propsComponent) => (
            <MovieDetailsWrraped
              {...propsComponent}
              films={films}
              authStatus={authStatus}
            />
          )}
        >
        </Route>
        <PrivateRoute
          exact
          path={AppRoute.MY_LIST}
          render={() => {
            return (
              <MyList />
            );
          }}
        />
        <PrivateRoute
          exact
          path={`${AppRoute.FILM}/:id${AppRoute.ADD_REVIEW}`}
          render={(propsComponent) => {
            return (
              <AddReviewWrraped
                {...propsComponent}
                films={films}
                onSubmitAddReview={onSubmitAddReview}
              />
            );
          }}
        />
        <Route exact path={`${AppRoute.PLAYER}/:id`}
          render={(propsComponent) => (
            <VideoFullScreenWrraped
              {...propsComponent}
              films={films}
            />
          )}
        />
        <Route
          render={() => {
            return (
              <PageWarring>
                <NotFoundMessage />
              </PageWarring>
            );
          }}
        />
      </Switch>
    </Router>
  );
};

const mapStateToProps = (state) => {
  return {
    films: getFilms(state),
    promoFilm: getPromoFilm(state),
    authStatus: getAuthStatus(state),
    userData: getUserData(state),
    genres: getUniqueGeners(state),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onSubmitAuth(authData: string, onSuccess, onError) {
      dispatch(UserOperation.login(authData, onSuccess, onError));
    },

    onSubmitAddReview(filmId, newComment, onSuccess, onError) {
      dispatch(DataOperation.addNewComment(filmId, newComment, onSuccess, onError));
    }
  };
};

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
