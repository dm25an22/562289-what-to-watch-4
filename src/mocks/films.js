
const COUNT_FILMS = 1;

const text = `Lorem Ipsum is simply dummy text of the printing and typesetting industry.
 Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer
  took a galley of type and scrambled it to make a type specimen book.`;

const films = [
  {
    title: `Fantastic Beasts`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Kids & Family`
  },
  {
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `Dramas`
  },
  {
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Revenant`,
    src: `img/revenant.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Johnny English`,
    src: `img/johnny-english.jpg`,
    genre: `Kids & Family`
  },
  {
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Seven Years in Tibet`,
    src: `img/seven-years-in-tibet.jpg`,
    genre: `Documentary`
  },
  {
    title: `Snatch`,
    src: `img/snatch.jpg`,
    genre: `Romance`
  },
  {
    title: `Mindhunter`,
    src: `img/mindhunter.jpg`,
    genre: `Horror`
  },
  {
    title: `Pulp Fiction`,
    src: `img/pulp-fiction.jpg`,
    genre: `Crime`
  },
  {
    title: `What we do in the shadows`,
    src: `img/what-we-do-in-the-shadows.jpg`,
    genre: `Sci-Fi`
  },
  {
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `Comedies`
  }
];

const actorsList = [
  `Robert Agnew`,
  `Jonathan Ahdout`,
  `Waris Ahluwalia`,
  `Ahmed Ahmed`,
  `Philip Ahn`,
  `Charles Aidman`,
  `Danny Aiello`,
  `Danny Aiello`,
  `Liam Aiken`,
  `Alan Aisenberg`,
  `Spottiswoode Aitken`,
  `Franklyn Ajaye`,
  `Ayad Akhtar`,
  `Gbenga Akinnagbe`,
  `Claude Akins`,
  `Marc Alaimo`,
  `Mark Alan`,
  `Matthew Alan`,
  `Rico Alaniz`,
  `Carlos Alazraqui`,
  `Luis Alberni`,
  `Devin Albert`,
  `Eddie Albert`,
];

const producers = [
  `Edward Janney`,
  `Sabra Jones`,
  `Tom Ruegger`,
  `Ron Senkowski`,
  `Kelly Leonard`,
  `Matthew Daniels`,
  `Olivia Luccardi`,
  `P. J. Wolfson`,
  `Yance Ford`,
  `Allen Estrin`,
  `Jenno Topping`,
];

const previews = [
  `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
];

const getRandomNumber = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

const getRandomRating = () => {
  const firstNumber = getRandomNumber(0, 10);
  return Number(`${firstNumber}.${firstNumber !== 10 && firstNumber !== 0 ? getRandomNumber(0, 9) : 0}`);
};

const getDescriptionRating = (rating) => {

  switch (true) {
    case rating < 3:
      return `Bad`;

    case rating >= 3 && rating < 5:
      return `Normal`;

    case rating >= 5 && rating < 8:
      return `Good`;

    case rating >= 8 && rating < 10:
      return `Very good`;

    case rating >= 10:
      return `Awesome`;
  }

  return `Unknown`;
};

const getRandomItem = (arr) => {
  return arr[getRandomNumber(0, arr.length - 1)];
};

const getRandomArrWithUniqeItems = (arr) => {
  const copyArr = [...arr];
  const result = [];
  for (let i = 0; i < getRandomNumber(2, arr.length - 1); i++) {
    result.push(copyArr.splice(getRandomNumber(0, copyArr.length - 1), 1)[0]);
  }

  return result;
};


const generateMock = () => {
  const ratingNumber = getRandomRating();
  const ratingString = getDescriptionRating(ratingNumber);
  const randomFilm = getRandomItem(films);

  return {
    title: randomFilm.title,
    moviePoster: `https://picsum.photos/300/450`,
    bigPoster: `https://picsum.photos/1900/550`,
    smallCardImg: randomFilm.src,
    genre: randomFilm.genre,
    year: getRandomNumber(1995, 2017),
    description: text,
    preview: getRandomItem(previews),
    rating: ratingNumber,
    descriptionRating: ratingString,
    quantityVotes: getRandomNumber(5, 300),
    producer: getRandomItem(producers),
    listActors: getRandomArrWithUniqeItems(actorsList)
  };
};

const createMock = (countFilms) => {
  return new Array(countFilms)
    .fill()
    .map(generateMock);
};

export default createMock(COUNT_FILMS);

