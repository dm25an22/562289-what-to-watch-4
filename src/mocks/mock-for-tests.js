const mockFilms = [
  {
    title: `Fantastic Beasts`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 1,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    title: `Title`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/bohemian-rhapsody.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 2,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    title: `Title2`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Action`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/macbeth.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 3,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
];

const mockPromo = {
  title: `Promo`,
  moviePoster: `film.poster_image`,
  bigPoster: `film.background_image`,
  smallCardImg: `film.preview_image`,
  genre: `film.genre`,
  year: 1998,
  description: `film.description`,
  preview: `film.preview_video_link`,
  rating: 3.2,
  descriptionRating: `Bad`,
  quantityVotes: 1290,
  producer: `film.director`,
  listActors: [`1`, `2`, `3`],
  runTime: 120,
  videoLink: `film.video_link`,
  isFavorite: true,
  backgroundColor: `red`,
  id: 4,
};

const userDataMock = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

const noop = () => {};

const commentsMock = [
  {
    "id": 1,
    "user": {
      "id": 4,
      "name": `Kate Muir`
    },
    "rating": 8.9,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  },
  {
    "id": 2,
    "user": {
      "id": 4,
      "name": `Kate Muir`
    },
    "rating": 2.0,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  },
  {
    "id": 3,
    "user": {
      "id": 4,
      "name": `Kate Muir`
    },
    "rating": 4.3,
    "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    "date": `2019-05-08T14:13:56.569Z`
  }
];

export {
  mockFilms,
  mockPromo,
  userDataMock,
  noop,
  commentsMock
};
