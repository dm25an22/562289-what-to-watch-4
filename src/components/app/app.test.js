import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";
import {mockFilms, userDataMock} from "../../mocks/mock-for-tests";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space";

const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};

const mockStore = configureStore({});

const promo = {
  id: 1,
  title: `Promo`,
  moviePoster: `film.poster_image`,
  bigPoster: `film.background_image`,
  smallCardImg: `film.preview_image`,
  genre: `film.genre`,
  year: 1998,
  description: `film.description`,
  preview: `film.preview_video_link`,
  rating: 3.2,
  descriptionRating: `Bad`,
  quantityVotes: 1290,
  producer: `film.director`,
  listActors: [`1`, `2`, `3`],
  runTime: 120,
  videoLink: `film.video_link`,
  isFavorite: true,
  backgroundColor: `red`
};

it(`renders App component without authorization user`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      films: mockFilms,
      promoFilm: promo,
      favorites: [promo]
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      userData: null
    }
  });

  window.scrollTo = jest.fn();

  const tree = renderer.create(
      <Provider store={store}>
        <App
          onSubmitAddReview={() => {}}
          films={mockFilms}
          promoFilm={promo}
          authStatus={AuthorizationStatus.NO_AUTH}
          onSubmitAuth={() => {}}
          loadFavoriteList={() => {}}
          genres={[`All genre`, `Drama`, `Action`]}
        />
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders App component with authorization user`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      films: mockFilms,
      promoFilm: promo,
      favorites: [promo]
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      userData: userDataMock
    }
  });

  window.scrollTo = jest.fn();

  const tree = renderer.create(
      <Provider store={store}>
        <App
          onSubmitAddReview={() => {}}
          films={mockFilms}
          promoFilm={promo}
          authStatus={AuthorizationStatus.AUTH}
          onSubmitAuth={() => {}}
          loadFavoriteList={() => {}}
          genres={[`All genre`, `Drama`, `Action`]}
        />
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});


