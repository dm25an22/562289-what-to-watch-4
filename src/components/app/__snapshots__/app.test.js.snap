// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component with authorization user 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="film.background_image"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link undefined"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="https://htmlacademy-react-3.appspot.comimg/1.png"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="film.poster_image"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Promo
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              film.genre
            </span>
            <span
              className="movie-card__year"
            >
              1998
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genre
          </a>
        </li>
        <li
          className="catalog__genres-item false"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Action
          </a>
        </li>
        <li
          className="catalog__genres-item false"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              height="175px"
              loop={true}
              muted={true}
              width="280px"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/1"
              onClick={[Function]}
            >
              Fantastic Beasts
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              height="175px"
              loop={true}
              muted={true}
              width="280px"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/2"
              onClick={[Function]}
            >
              Title
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              height="175px"
              loop={true}
              muted={true}
              width="280px"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/3"
              onClick={[Function]}
            >
              Title2
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`renders App component without authorization user 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="film.background_image"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link undefined"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="film.poster_image"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Promo
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              film.genre
            </span>
            <span
              className="movie-card__year"
            >
              1998
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genre
          </a>
        </li>
        <li
          className="catalog__genres-item false"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Action
          </a>
        </li>
        <li
          className="catalog__genres-item false"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              height="175px"
              loop={true}
              muted={true}
              width="280px"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/1"
              onClick={[Function]}
            >
              Fantastic Beasts
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              height="175px"
              loop={true}
              muted={true}
              width="280px"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/2"
              onClick={[Function]}
            >
              Title
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              height="175px"
              loop={true}
              muted={true}
              width="280px"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/3"
              onClick={[Function]}
            >
              Title2
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
